cep-20.2.md

```diff
@@ -0,0 +1,650 @@
+</table>
+
+## Abstract
+
+We propose a new recipe format that is heavily inspired by conda-build. The main
+change is a pure YAML format without arbitrary Jinja or comments with semantic
+meaning.
+
+## Motivation
+
+The conda-build format is currently under-specified. For the new format, we try to list all values
+and types in a single document and document them. This is part of a larger effort (see [](/cep-20.1.md) for the new YAML syntax).
+
+## History
+
+A discussion was started on what a new recipe spec could or should look like.
+The fragments of this discussion can be found here:
+https://github.com/mamba-org/conda-specs/blob/master/proposed_specs/recipe.md
+The reason for a new spec are:
+
+- Make it easier to parse ("pure yaml"). conda-build uses a mix of comments and
+  jinja to achieve a great deal of flexibility, but it's hard to parse the
+  recipe with a computer
+- iron out some inconsistencies around multiple outputs (build vs. build/script
+  and more)
+- remove any need for recursive parsing & solving
+- cater to needs for automation and dependency tree analysis via a determinstic
+  format
+
+## Major differences with conda-build
+
+Outputs and top-level package are mutually exclusive, and outputs have exactly the same structure as top-level keys.
+If outputs exist, the top-level keys are 'merged' with the output keys (e.g. for the about section).
+
+# Format
+
+## Context section
+
+```yaml
+# The context dictionary defines arbitrary key-value pairs for Jinja interpolation
+# and replaces the {% set ... %} commands commonly used in recipes
+context:
+  variable: test
+  # note that we can reference previous values
+  other_variable: test_${{ variable }}
+```
+
+## Package section
+
+```yaml
+package:
+  # The name of the package
+  name: string
+  # The version of the package, following the conda version spec
+  # Note that versions are always strings (1.23 is _not_ a valid version since it is a `float`. Needs to be quoted.)
+  version: string
+```
+
+## Build section
+
+```yaml
+build:
+    # the build number
+    number: Option<integer>, defaults to 0
+    # the build string. This is usually ommited
+    string: Option<string>, defaults to a build string made from "package hash & build number"
+
+    # A list of conditions under which to skip the build of this package
+    skip: [list of expressions]
+
+    force_use_keys: [string]
+    force_ignore_keys: [string]
+
+    # wether the package is a noarch package, and if yes, wether it is "generic" or "python"
+    noarch: Option<"generic" | "python">
+
+    # Only valid if `noarch: python` - list of all entry points of the package
+    # PythonEntryPoint: `bsdiff4 = bsdiff4.cli:main_bsdiff4`
+    entry_points: [PythonEntryPoint]
+
+    # ?
+    osx_is_app: bool (default false)
+
+    # used on conda-forge, still needed?
+    preserve_egg_dir: bool (default false)
+
+    # only used as a hack to down-prioritize packages
+    track_features: [string]
```

Comment:
Is it a good moment to rename this (even if we use `track_features` under the hood)?

Reply:
probably!

Reply:
Do we have an actual proposal for what to call the next thing to replace track features? Maybe that has to come first?

Reply:
I would propose `priority` or `downprioritize` and then an integer (positive or negative)